docs/modules/extensions/pages/index.adoc

```diff
@@ -1,51 +1,76 @@
 = Extensions
 :url-exten-lab: https://github.com/asciidoctor/asciidoctor-extensions-lab
 
-Extensions are central to the success of AsciiDoc because they open up the language to new use cases.
-Asciidoctor provides an extension API that offers a superset of extension points.
-As a result, extensions in Asciidoctor are easy to write, powerful, and simple to distribute.
+Extensions are central to the power of AsciiDoc because they open up the language to new use cases.
+Asciidoctor provides an extension API that offers a set of extension points.
+Extensions in Asciidoctor are easy to write, powerful, and simple to distribute.
 
-Asciidoctor also allows extensions to be written using the full power of a programming language (whether it be Ruby, Java, Groovy or JavaScript).
-You don't have to shave yaks to get the functionality you want, and you can distribute the extension using defacto-standard packaging mechanisms like RubyGems or JARs.
+Asciidoctor also allows extensions to be written using the full power of the programming language of your asciidoctor variant: Ruby, Java or other JVM based languages, or Javascript.
+An extension may be included in custom code or packaged and distributed using a language-standard mechanism such as Gems, Jars/Maven, or npm packages.
 
-== Available extension points
+There are two general types of processors:
 
-Asciidoctor provides the following extension points:
+Document Processor::
+  A document processor does not have an explicit invocation in the document.
+Except for `IncludeProcessor`, a document processor is invoked at the appropriate stage of processing with the entire document.
```

Comment:
I'm going to experiment to see if there's a way to interleave a description of the processing stages and the processor types.  It may take a while :-)

Reply:
I'm reasonably happy with what I came up with for an interleaved description.

Reply:
Yes, it looks really great, much better than my attempts :grinning:

Reply:
I agree we're not there yet.  I think the crucial distinctions are that a doc processor operates on text, and has no invocation, and a syntax processor operates on the DOM/AST tree.